datalad/core/local/run.py

```diff
@@ -823,30 +828,34 @@ def run_command(cmd, dataset=None, inputs=None, outputs=None, expand=None,
     if outputs_to_save is not None and use_sidecar:
         outputs_to_save.append(record_path)
     do_save = outputs_to_save is None or outputs_to_save
+    msg_path = None
     if not rerun_info and cmd_exitcode:
         if do_save:
             repo = ds.repo
-            msg_path = repo.dot_git / "COMMIT_EDITMSG"
-            msg_path.write_text(msg)
-            if datalad.get_apimode() == 'python':
-                help = f"\"Dataset('{ds.path}').save(path='.', " \
-                       "recursive=True, message_file='%s')\""
-            else:
-                help = "'datalad save -d . -r -F %s'"
-            lgr.info("The command had a non-zero exit code. "
-                     "If this is expected, you can save the changes with "
-                     f"{help}",
-                     # shorten to the relative path for a more concise message
-                     msg_path.relative_to(ds.pathobj))
-            if repo.dirty and not explicit:
-                # Give clean-up hints if a formerly clean repo is left dirty
-                lgr.info(
-                    "The commands 'git clean -di' and/or 'git reset' "
-                    "could be used to get to a clean dataset state again. "
-                    "Consult their man pages for more information."
-                )
-        raise exc
-    elif do_save:
+            msg_path = relpath(opj(str(repo.dot_git), "COMMIT_EDITMSG"))
```

Comment:
Why turn the `Path` in `repo.dot_git` into a str to then use it with `os.path` rather than just using `pathlib`?

Reply:
This is the original change that Kyle made. Any suggested changes are most welcome!

Reply:
It actually seems to be redundant with the post-processing done in the result renderer. So I guess I resolved the merge conflicts in a suboptimal fashion. Will correct.

Reply:
I have addressed this now. It still needs to be recoded from Repo to Dataset semantics, though.